src/groupeddataframe/fastaggregates.jl

```diff
@@ -193,6 +194,12 @@ function groupreduce!(res::AbstractVector, f, op, condf, adjust, checkempty::Boo
             end
         end
     end
+    # Eliminate the argument-type diversity that we no longer need
+    return _groupreduce!(res, counts, n, op === min || op === max, adjust, checkempty, incol)
+end
+
+function _groupreduce!(res::AbstractVector, counts::Union{Nothing,Vector{Int}}, n::Int, opminormax::Bool, adjust, checkempty::Bool,
+                      incol::AbstractVector)
```

Comment:
If you just pass `op`, no specialization should happen on it since it's not called by the function, right? I'd find it clearer.

Also I'd use a more explicit name  like `groupreduce_fixres` or something.